### PR TITLE
TMA-843: avoid abuse of obj resource in partial md import export

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/models/project.rb [hunk 1]
```diff
@@ -1259,32 +1252,21 @@ def partial_md_export(objects, options = {})
...
-              }
-            rescue GoodData::ObjectsImportError => e
-              {
-                project: target_project,
-                result: false,
-                reason: e.message
-              }
-            end
+            target_project = client.projects(proj)
+            target_project.objects_import(token, options)
```
Comment (yenkins) [inline lib/gooddata/models/project.rb]: Calls 'target_project.objects_import(token, options)' 2 times (DuplicateMethodCall)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/932/ : SUCCESS in 27s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/791/ : SUCCESS in 1m 53s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1059/ : SUCCESS in 1m 51s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/337/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/796/ : SUCCESS in 2m 09s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1064/ : SUCCESS in 36s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1318/ : FAILURE in 55m 30s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1081/ : SUCCESS in 36m 04s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/340/ : FAILURE in 18s
- gooddata-ruby-pronto-code-review-zuul : SKIPPED
- gooddata-ruby-unit-tests-zuul : SKIPPED
- gooddata-ruby-integration-tests-zuul : SKIPPED
- gooddata-ruby-lcm-integration-tests-zuul : SKIPPED

**Comment by Hahihula** [comment]
ok to merge

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/341/ : SUCCESS in 17s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/802/ : SUCCESS in 2m 16s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1070/ : SUCCESS in 2m 13s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1322/ : FAILURE in 1h 28m 19s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1085/ : FAILURE in 16m 27s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/342/ : SUCCESS in 24s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/803/ : SUCCESS in 38s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1071/ : SUCCESS in 2m 01s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1324/ : FAILURE in 19m 25s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1087/ : UNSTABLE in 5m 13s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/343/ : SUCCESS in 18s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/810/ : SUCCESS in 2m 22s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1078/ : SUCCESS in 2m 00s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1326/ : ABORTED in 2h 00m 02s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1089/ : UNSTABLE in 39m 04s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/1/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/811/ : SUCCESS in 2m 25s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1079/ : SUCCESS in 3m 55s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1328/ : FAILURE in 54m 50s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1091/ : SUCCESS in 33m 45s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/4/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/820/ : SUCCESS in 43s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1088/ : SUCCESS in 35s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1332/ : FAILURE in 52m 43s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1095/ : SUCCESS in 33m 47s

File: lib/gooddata/models/project.rb [hunk 2]
```diff
@@ -1178,16 +1178,9 @@ def objects_export(objs, options = {})
...
-      if objs.any? { |_, obj| obj.nil? }
-        object = objs.select { |_, obj| obj.nil? }.map { |o, _| o }.join(', ')
-        error_message = "Exporting objects failed with messages. " \
-                        "Object #{object} could not be found."
-        fail ObjectsExportError, error_message
-      end
       export_payload = {
         :partialMDExport => {
-          :uris => objs.map { |_, obj| obj.uri },
+          :uris => objs,
```
Comment (panjan) [inline lib/gooddata/models/project.rb]: Does this mean we expect objs to be strings? In that case the documentation doesn't apply: https://github.com/gooddata/gooddata-ruby/pull/1152/files#diff-e097692ced3388707c4489714a5d7cd9R1168

What happens when objs is an array of mdobjects? Do we have a test for it? objects_export is a publicly documented method so we must be sure there is no regression.

Comment (kubamahnert) [reply lib/gooddata/models/project.rb]: fixed

File: lib/gooddata/models/project.rb [hunk 3]
```diff
@@ -1172,22 +1172,15 @@ def objects(id, opts = {})
     # @return [String] Returns token that you can use as input for object_import
     def objects_export(objs, options = {})
       fail 'Nothing to migrate. You have to pass list of objects, ids or uris that you would like to migrate' if objs.nil?
-      objs = Array(objs)
+      objs = Array(objs).map { |o| o.respond_to?(:uri) ? o.uri : o }
```
Comment (yenkins) [inline lib/gooddata/models/project.rb]: Manually dispatches method call (ManualDispatch)

Comment (yenkins) [inline lib/gooddata/models/project.rb]: Has the variable name 'o' (UncommunicativeVariableName)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/20/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/826/ : SUCCESS in 2m 52s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1094/ : SUCCESS in 52s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/12/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/837/ : SUCCESS in 2m 22s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1105/ : SUCCESS in 2m 32s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1343/ : SUCCESS in 55m 16s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1106/ : SUCCESS in 34m 10s

**Pull request merged**